cairo/src/font/user_fonts.rs

```diff
@@ -0,0 +1,206 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use super::{FontExtents, FontFace, ScaledFont, TextCluster, TextClusterFlags, TextExtents};
+use crate::utils::status_to_result;
+use crate::{Context, Error, Glyph};
+
+pub struct UserFontFace(FontFace);
+
+impl UserFontFace {
+    #[doc(alias = "cairo_user_font_face_create")]
+    pub fn create() -> Result<FontFace, Error> {
+        let font_face = unsafe { FontFace::from_raw_full(ffi::cairo_user_font_face_create()) };
+        let status = unsafe { ffi::cairo_font_face_status(font_face.to_raw_none()) };
+        status_to_result(status)?;
+        Ok(font_face)
+    }
+
+    #[doc(alias = "cairo_user_font_face_set_init_func")]
+    pub fn set_init_func<F>(&self, func: F)
+    where
+        F: Fn(&mut ScaledFont, &Context) -> Result<FontExtents, Error> + Send + Sync + 'static,
+    {
+        static INIT_FUNC: once_cell::sync::OnceCell<
+            Box<dyn Fn(&mut ScaledFont, &Context) -> Result<FontExtents, Error> + Send + Sync>,
+        > = once_cell::sync::OnceCell::new();
+        if INIT_FUNC.set(Box::new(func)).is_err() {
+            panic!("Init func can only be set once")
+        }
+        unsafe extern "C" fn init_trampoline(
+            scaled_font: *mut ffi::cairo_scaled_font_t,
+            cr: *mut ffi::cairo_t,
+            extents: *mut ffi::cairo_font_extents_t,
+        ) -> ffi::cairo_status_t {
+            let init_func = INIT_FUNC.get().unwrap();
+            match init_func(
+                &mut ScaledFont::from_raw_none(scaled_font),
+                &Context::from_raw_none(cr),
+            ) {
+                Err(err) => err.into(),
+                Ok(font_extents) => {
+                    *extents = std::mem::transmute(font_extents);
```

Comment:
```suggestion
                    *extents = font_extents.0;
```

Don't use `transmute` if it can be avoided

Reply:
.0 is private, can not access at there.

Reply:
Dose this could work?
```
*extents = font_extents as _;
```

Reply:
It would make sense to add an `impl From<FontExtends> for ffi::cairo_font_extends_t`